IdentityCore/src/cache/MSIDKeychainTokenCache.m

```diff
@@ -427,16 +427,38 @@ - (MSIDAccountMetadataCacheItem *)accountMetadataWithKey:(MSIDCacheKey *)key
                                                  context:(id<MSIDRequestContext>)context
                                                    error:(NSError **)error
 {
-    NSArray *items = [self itemsWithKey:key context:context error:error];
+    NSArray *metadataItems = [self accountsMetadataWithKey:key serializer:serializer context:context error:error];
+    if (!metadataItems) return nil;
     
-    if (!items || items.count < 1)
+    if (metadataItems.count < 1)
     {
         MSID_LOG_WITH_CTX(MSIDLogLevelWarning,context, @"Found no metadata item.");
         return nil;
     }
     
-    NSData *itemData = [items[0] objectForKey:(id)kSecValueData];
-    return (MSIDAccountMetadataCacheItem *)[serializer deserializeCacheItem:itemData ofClass:[MSIDAccountMetadataCacheItem class]];
+    return metadataItems[0];
+}
+
+- (NSArray<MSIDAccountMetadataCacheItem *> *)accountsMetadataWithKey:(MSIDCacheKey *)key
```

Comment:
can we use cacheItemsWithKey here?

Reply:
:+1: didn't notice that help method there :P now changed.